powertools-batch/src/main/java/software/amazon/lambda/powertools/batch/handler/BatchMessageHandler.java

```diff
@@ -33,6 +33,21 @@ public interface BatchMessageHandler<E, R> {
      * @param context The lambda context
      * @return A partial batch response
      */
-    public abstract R processBatch(E event, Context context);
+    R processBatch(E event, Context context);
 
+    /**
+     * Processes the given batch in parallel returning a partial batch
+     * response indicating the success and failure of individual
+     * messages within the batch. <br/>
+     * Note that parallel processing is not always better than sequential processing,
+     * and you should benchmark your code to determine the best approach for your use case. <br/>
+     * Also note that to get more threads available (more vCPUs),
+     * you need to increase the amount of memory allocated to your Lambda function. <br/>
```

Comment:
Or, override the thread pool on the builder , if we add this

Reply:
I think it's a bad idea...

Reply:
@jeromevdl can you elaborate?

Reply:
https://github.com/aws-powertools/powertools-lambda-java/pull/1620#discussion_r1654331875